### PR TITLE
Update README.md

### LINKED ISSUE #54: Add a section about building and modifying the read-the-docs documentation locally?
I'm not very well versed in building Sphinx and read-the-docs documentation, and struggled a bit with creating a local environment.

I ended up using the make script `make html` and `pip install` all the missing modules, ~~then [sphinx-serve](https://pypi.org/project/sphinx-serve/)~~.

**Edit:** The package [sphinx-autobuild](https://pypi.org/project/sphinx-autobuild/) suggested by Pete below is much better than the one I suggested and is being actively maintenained. It will automatically reload the page in a browser when changes the source file is saved.

I could start a pull request and describe how I got my environment to work, if you'd like?

### PR DESCRIPTION
**Pull request by alanocallaghan**
Resolves #54

### CODE REVIEW

**Review by finglis — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -4,13 +4,27 @@ This contains the source for QuPath's documentation, hosted at https://qupath.re
...
+```bash
+python -m venv ./venv
+. ./venv/bin/activate
+```
+
+or using conda/mamba:
+
+```bash
+conda env create -n qupath-docs
+conda activate qupath-docs
```
Comment (petebankhead) [inline README.md:18]: You'll need to install Python here

File: README.md [hunk 2]
```diff
@@ -4,13 +4,27 @@ This contains the source for QuPath's documentation, hosted at https://qupath.re
...
+```
+
+Then install the requirements for this repo:
+
+```bash
+pip install -r requirements.txt
+```
+
+You'll also need the command line tool `Make` (e.g., [GNU Make](https://www.gnu.org/software/make/)).
```
Comment (petebankhead) [inline README.md:27]: Does this need to be installed separately on Windows?
I think macOS/Linux should generally have it - right?

File: README.md (hunk 1 above)
Comment (alanocallaghan) [reply README.md:18]: I thought conda envs generally come with python and pip?

File: README.md (hunk 2 above)
Comment (alanocallaghan) [reply README.md:27]: I would have to check later, I suspect you can install it with conda so it'd be easiest to point people that way

Yeah afaik mac and linux will always have some form of make

File: README.md (hunk 1 above)
Comment (petebankhead) [reply README.md:18]: It seems not, unless you use
```
mamba create -n qupath-docs python
```
May be safer to specify version, e.g.
```
mamba create -n qupath-docs python=3.12
```

**Review by petebankhead — APPROVED** [review]

**Comment by alanocallaghan** [comment]
Added sphinx-autobuild instructions, feel free to test

**Comment by petebankhead** [comment]
I'm not sure if we want these in the `requirements.txt`, since they aren't required - why not just leave it up to the user if they want to install `sphinx-autobuild` on their own environment?

**Comment by petebankhead** [comment]
The `requirements.txt` is really intended for readthedocs: https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html#use-a-requirements-file-for-python-dependencies

Although it doesn't currently specify a `sphinx` version... I suspect it was written far enough in the past that the readthedocs docs could have changed in the interim, and we are some way off best practice.

**Comment by alanocallaghan** [comment]
Yeah fair enough, removed from reqs but left in the readme

**Pull request merged**